tests/suites/test_suite_bignum_mod.function

```diff
@@ -102,6 +103,155 @@ exit:
 
 /* BEGIN MERGE SLOT 2 */
 
+/* BEGIN_CASE */
+void mpi_mod_mul( char * input_A,
+                  char * input_B,
+                  char * input_N,
+                  char * result )
+{
+    mbedtls_mpi_uint *X = NULL;
+
+    mbedtls_mpi_mod_modulus m;
+    mbedtls_mpi_mod_modulus_init( &m );
+
+    TEST_EQUAL( test_read_modulus( &m, MBEDTLS_MPI_MOD_REP_MONTGOMERY, input_N ),
+                0 );
+
+    mbedtls_mpi_mod_residue rA;
```

Comment:
Those structures (`rA`, `rB`, `rR` )should be declared before the first test.

Reply:
Since these structures initialized by a function and not by an assignment it won't cause any problem. The declaration can be anywhere before the first use and the compiler will handle it.

Reply:
This is for the sake of consistency rather than compiler issues. Having a look at the test suites,  the declaration's in each function come before the functional code which is the tests.

Reply:
Err, no, that's not right - the `mbedtls_mpi_mod_residue`s need to be declared and initialised to `{ NULL, 0 }` before the first `TEST_xxx()` that might `goto exit`, since the code at `exit:` uses the values in the `p` members. And then the `->p`s need to be freed before the residues are released.

i.e. this order (not complete):

```
    mbedtls_mpi_mod_residue rA = { NULL, 0 };
    mbedtls_mpi_mod_residue rB = { NULL, 0 };
    mbedtls_mpi_mod_residue rR = { NULL, 0 };
    :
    : first TEST_EQUAL()
    :
    TEST_EQUAL( test_read_residue( &rA, &m, input_A, 0 ), 0 );
    TEST_EQUAL( test_read_residue( &rB, &m, input_B, 0 ), 0 );
    :
exit:
    mbedtls_free( rA.p );
    mbedtls_free( rB.p );
    mbedtls_free( rR.p );

    mbedtls_mpi_mod_residue_release( &rA );
    mbedtls_mpi_mod_residue_release( &rB );
    mbedtls_mpi_mod_residue_release( &rR );
    :
```